halva/src/main/java/io/soabase/halva/comprehension/MonadicFor.java

```diff
@@ -0,0 +1,43 @@
+package io.soabase.halva.comprehension;
+
+import io.soabase.halva.any.AnyVal;
+
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+import java.util.function.Function;
+
+/** Interface class for the non-type-safe version "for comprehension" for Monadic classes

```

Comment:
It's non type safe in 3 ways:
* the initialization needs a type less cast (not really non type safe just
ugly!),
* the setting of any vars is non type safe (which I think yours is also?
But only because you had 2 types T and R, I was going to ask why you didn't
just have T?)
* the yield return want type safe

(All 3 of these issues are fixed by the option/future/stream
specializations)
On Jun 23, 2016 8:16 AM, "Jordan Zimmerman" <notifications@github.com>
wrote:

In halva/src/main/java/io/soabase/halva/comprehension/MonadicFor.java
<https://github.com/Randgalt/halva/pull/16#discussion_r68223636>:

> @@ -0,0 +1,43 @@
> +package io.soabase.halva.comprehension;
> +
> +import io.soabase.halva.any.AnyVal;
> +
> +import java.util.function.Predicate;
> +import java.util.function.Supplier;
> +import java.util.function.Function;
> +
> +/** Interface class for the non-type-safe version "for comprehension" for Monadic classes

Is it really non-type-safe? I'm not sure. I think it's just as type safe as
my implementation.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/Randgalt/halva/pull/16/files/cf3898b0df153276053c9c231566874716e89a44#r68223636>,
or mute the thread
<https://github.com/notifications/unsubscribe/AQwn4Pf47_YQ-_DZhpw2Zd0NvQ2WfJKQks5qOnkqgaJpZM4I8CC2>
.